frontend/src/hooks/useActions.ts

```diff
@@ -193,14 +179,20 @@ const useActions = (registerHotkeys = false) => {
     DELETE: {
       hotkeys: [{ key: 'Delete' }, { key: 'Backspace' }],
       handler: () => {
-        const selectionState = useSelectionStore.getState()
-        const selectedStateIDs = selectionState.selectedStates
-        const selectedTransitionIDs = selectionState.selectedTransitions
-        const selectedCommentIDs = selectionState.selectedComments
-        if (selectedStateIDs.length > 0 || selectedTransitionIDs.length > 0 || selectedCommentIDs.length > 0) {
-          removeStates(selectedStateIDs)
-          removeTransitions(selectedTransitionIDs)
-          removeComments(selectedCommentIDs)
+        // If a template is selected and nothing else is, delete the template
+        if (template !== null && selectedCommentsIds.length === 0 && selectedStatesIds.length === 0 && selectedTransitionsIds.length === 0) {
+          if (window.confirm(`Are you sure you want to delete your template '${template.name}'?`)) {
+            deleteTemplate(template._id)
+            setTemplate(null)
+            setTool('cursor')
+            return
+          }
+        }
+        // Otherwise, delete selection
+        if (selectedStatesIds.length > 0 || selectedTransitionsIds.length > 0 || selectedCommentsIds.length > 0) {
```

Comment:
```suggestion
        else if (selectedStatesIds.length > 0 || selectedTransitionsIds.length > 0 || selectedCommentsIds.length > 0) {
        // Otherwise, delete selection 
```
imo this is better since it shows that only one branch can run